tensorflow_addons/optimizers/gradient_accumulator.py

```diff
@@ -39,26 +45,36 @@ def __call__(self, grads_and_vars, *args, **kwargs):
         step_inc_op = self.step.assign_add(1, read_value=False)
 
         with tf.control_dependencies([step_inc_op]):
-            can_apply = tf.cast(self.step % self.accu_steps == 0, dtype=self.variable_dtype)
-            accumulate = tf.cast(self.step % tf.constant(self.accu_steps + 1, dtype=self.step.dtype) != 0, dtype=self.variable_dtype)
+            can_apply = tf.cast(
+                self.step % self.accu_steps == 0, dtype=self.variable_dtype
+            )
+            accumulate = tf.cast(
+                self.step % (self.accu_steps + 1) != 0, dtype=self.variable_dtype
+            )
 
         accum_ops = list()
         for grad, var in grads_and_vars:
+
+            # Get the accumulated gradient
             grad_accum = accu_gradients[var.ref()] * accumulate
+
             if isinstance(grad, tf.IndexedSlices):
                 added = tf.IndexedSlices(
-                    values=grad.values + tf.gather_nd(grad_accum, grad.indices[..., None]),
+                    values=grad.values + tf.gather_nd(grad_accum, grad.indices),
                     indices=grad.indices,
-                    dense_shape=grad.dense_shape
+                    dense_shape=grad.dense_shape,
                 )
                 accu_op = accu_gradients[var.ref()].scatter_update(added)
```

Comment:
scatter_add

Reply:
No, this has to be an update. Same for https://github.com/fsx950223/addons/pull/2#discussion_r672882006 which as to be an assign. `grad_accum` is accumulating all the values.